src/vm_id_handle.cc

```diff
@@ -19,41 +19,53 @@
 #include <mutex>
 #include <memory>
 #include <unordered_map>
+#include <vector>
 
 namespace proxy_wasm {
 
-std::unordered_map<std::string, std::weak_ptr<VmIdHandle>> global_vm_id_handles;
-std::vector<std::function<void(std::string_view vm_id)>> global_vm_id_handle_callbacks;
+std::unordered_map<std::string, std::weak_ptr<VmIdHandle>> *global_vm_id_handles = nullptr;
+std::vector<std::function<void(std::string_view vm_id)>> *global_vm_id_handle_callbacks = nullptr;
 std::mutex global_vm_id_handle_mutex;
```

Comment:
suspicious about this mutex for Windows test failure

Reply:
Yeah, I think mutex fails under the same fiasco.

Reply:
so should we fix https://github.com/proxy-wasm/proxy-wasm-cpp-host/blob/master/src/wasm.cc#L44 as well? though these are never access in the initialization phase since they are not accessed until createWasm is called

Reply:
No, those are fine. The issue here is `global_shared_data` that requires all the `vm_handle` objects.